csharp/src/Ice/Connection.cs

```diff
@@ -709,17 +688,17 @@ private async Task WaitForGoAwayAsync(SocketStream peerControlStream)
 
             async Task PerformGoAwayAsync((long Bidirectional, long Unidirectional) lastStreamIds, Exception exception)
             {
+                // Yield to ensure the code below is executed without the mutex locked (PerformGoAwayAsync is called
```

Comment:
The comment should explain why having the mutex locked is bad.

Reply:
I'll also remove it here.